libvcx/src/api_lib/api_handle/credential_def.rs

```diff
@@ -122,13 +135,16 @@ pub fn check_is_published(handle: u32) -> VcxResult<bool> {
     })
 }
 
-pub fn rotate_rev_reg_def(handle: u32, revocation_details: &str, new_tails_url: &str) -> VcxResult<String> {
+// todo: add 'publish revocation primitives' method
+pub fn rotate_rev_reg_def(handle: u32, revocation_details: &str) -> VcxResult<String> {
     CREDENTIALDEF_MAP.get_mut(handle, |s| {
         match &s.get_rev_reg_def()? {
             Some(_) => {
                 let revocation_details: RevocationDetails = serde_json::from_str(&revocation_details)
                     .map_err(|err| VcxError::from_msg(VcxErrorKind::SerializationError, format!("Failed to deserialize revocation details: {:?}, error: {:?}", revocation_details, err)))?;
-                let new_rev_reg = s.rotate_rev_reg(revocation_details, new_tails_url)?;
+                let new_rev_reg = s.rotate_rev_reg(revocation_details)?;
+                // todo: We update this cache, but where is it actaully utilised? didn't find any reads for reg_ids_cache
+                // if it is used, we should avoid inconsistent state between rotation and publishing
```

Comment:
We definitely use rev reg delta cache for batch revocations. I think keeping rev reg ids in a cache was part of a plan to expose API to get rev registries for a credential definition, which hasn't materialized yet...

Reply:
I removed the todo, however the comment in the code was correct - the thing is, it is caching rev reg ids (`update_rev_reg_ids_cache`), not deltas.